persistence-cassandra/core/src/main/scala/com/lightbend/lagom/internal/persistence/cassandra/ServiceLocatorHolder.scala

```diff
@@ -21,21 +25,36 @@ private[lagom] object ServiceLocatorHolder extends ExtensionId[ServiceLocatorHol
   override def lookup = ServiceLocatorHolder
 
   override def createExtension(system: ExtendedActorSystem): ServiceLocatorHolder =
-    new ServiceLocatorHolder
+    new ServiceLocatorHolder(system)
 }
 
-private[lagom] class ServiceLocatorHolder extends Extension {
+private[lagom] class ServiceLocatorHolder(system: ExtendedActorSystem) extends Extension {
   @volatile private var _serviceLocator: Option[ServiceLocatorAdapter] = None
 
-  def serviceLocator: Option[ServiceLocatorAdapter] = _serviceLocator
+  private val promisedServiceLocator = Promise[ServiceLocatorAdapter]()
+
+  private implicit val exCtx = system.dispatcher
+  private val delayed = akka.pattern.after(10.seconds, using = system.scheduler) {
```

Comment:
This timeout only controls this wait is not infinite. This timeout is not related to starting any service, it is related to building a valid Guice Injector (was a service locator class injected in less that 10 seconds?) . The time to build the Guice Injector should stay under 10 seconds and anything above that is an error. 

For the record, this timeout is a lot more generous than previous implementation where an unconfigurable polling loop only allowed for this to successfully complete in under 2 seconds.